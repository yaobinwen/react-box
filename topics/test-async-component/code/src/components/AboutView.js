import React, { Component } from 'react';
import './About.css';

class AboutView extends Component {
  render() {
    return (
      <div className="About">
        <p key="author" className="About-entry">
          Author: {this.props.author}
        </p>
        <p key="description" className="About-entry">
          Description: {this.props.description}
        </p>
      </div>
    );
  }
}

export default AboutView;