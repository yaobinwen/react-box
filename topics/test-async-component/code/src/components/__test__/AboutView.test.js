import React from 'react';
import renderer from 'react-test-renderer';
import AboutView from '../AboutView';

test('renders without crashing', () => {
  expect.assertions(1);

  let tree = renderer.create(
    <AboutView
      author='Test Author'
      description='Test Description'
    />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});