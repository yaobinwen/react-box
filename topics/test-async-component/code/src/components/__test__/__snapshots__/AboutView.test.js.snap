// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="About"
>
  <p
    className="About-entry"
  >
    Author: 
    Test Author
  </p>
  <p
    className="About-entry"
  >
    Description: 
    Test Description
  </p>
</div>
`;
